src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -561,41 +556,49 @@ private Set<ProgrammingExerciseTestCase> filterTestCasesForCurrentDate(Participa
 
     /**
      * Filters the test cases to only include the ones a student should be able to see.
-     * @param testCases all test cases of an exercise.
-     * @param isBeforeDueDate true, if the due date has not yet passed.
      * @return a set of test cases that are visible to the student.
      */
     private Set<ProgrammingExerciseTestCase> filterTestCasesForStudents(final Set<ProgrammingExerciseTestCase> testCases, boolean isBeforeDueDate) {
         return testCases.stream().filter(testCase -> !testCase.isInvisible()).filter(testCase -> !(isBeforeDueDate && testCase.isAfterDueDate())).collect(Collectors.toSet());
     }
 
+    private record ScoreCalculationData(ProgrammingExercise exercise, Result result, Set<ProgrammingExerciseTestCase> testCases,
+            Set<ProgrammingExerciseTestCase> successfulTestCases, List<Feedback> staticCodeAnalysisFeedback, double weightSum) {
+
+        ScoreCalculationData(ProgrammingExercise exercise, Result result, Set<ProgrammingExerciseTestCase> testCases, Set<ProgrammingExerciseTestCase> successfulTestCases,
+                List<Feedback> staticCodeAnalysisFeedback) {
+            this(exercise, result, testCases, successfulTestCases, staticCodeAnalysisFeedback, calculateWeightSum(testCases));
+        }
+
+        private static double calculateWeightSum(final Set<ProgrammingExerciseTestCase> allTests) {
+            return allTests.stream().filter(testCase -> !testCase.isInvisible()).mapToDouble(ProgrammingExerciseTestCase::getWeight).sum();
+        }
+
+        public Participation participation() {
```

Comment:
I would argue the participation does not belong directly to ScoreCalculationData. Therefore I would remove this getter and access it via result.getParticipation() in the rest of the code.

Reply:
We use participation to calculate the score. This is just a bundling of previous parameters that were used.